src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+  g_autofree gchar *summary_url = NULL;
+
+  if (url_override != NULL)
+    url = g_strdup (url_override);
+  else if (!ostree_repo_remote_get_url (repo, remote_name, &url, error))
+    return FALSE;
+
+  summary_url = g_strconcat (url, "/summary", NULL);
+  return commit_checksum_from_any_summary (repo,
+                                           remote_name,
+                                           ref,
+                                           summary_url,
+                                           cancellable,
+                                           out_checksum,
+                                           out_extensions,
+                                           error);
+}
+
+static gboolean
+fetch_commit_checksum (OstreeRepo *repo,
+                       GCancellable *cancellable,
+                       const gchar *remote_name,
+                       const gchar *ref,
+                       const gchar *url_override,
+                       gchar **out_checksum,
+                       EosExtensions **out_extensions,
+                       GError **error)
+{
+  if (commit_checksum_from_extensions_ref (repo,
+                                           cancellable,
+                                           remote_name,
+                                           ref,
+                                           url_override,
+                                           out_checksum,
+                                           out_extensions,
+                                           error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_extensions_summary (repo,
+                                               cancellable,
+                                               remote_name,
+                                               ref,
+                                               url_override,
+                                               out_checksum,
+                                               out_extensions,
+                                               error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_summary (repo,
+                                    cancellable,
+                                    remote_name,
+                                    ref,
+                                    url_override,
+                                    out_checksum,
+                                    out_extensions,
+                                    error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (url_override != NULL)
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s with URL %s",
+                 ref, remote_name, url_override);
+  else
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s",
+                 ref, remote_name);
```

Comment:
Shameless plug - https://bugzilla.gnome.org/show_bug.cgi?id=124022#c12 ;)

Anyway, I rather really hope that we will just use the refs file and drop the summary thing, so I won't have that problem. Hopefully that will be decided tomorrow. If we decide to keep it, then I'll come up with something.